pkg/types/installconfig.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import (
+	"net"
+
+	"github.com/pborman/uuid"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type InstallConfig struct {
+	// +optional
+	metav1.TypeMeta `json:",inline"`
+
+	metav1.ObjectMeta `json:"metadata"`
+
+	// ClusterID is the ID of the cluster.
+	ClusterID uuid.UUID `json:"clusterID"`
```

Comment:
I'd rather have this be a `[]byte` (or hex-encoded `string`?).  Is there a reason to preserve the `UUID` type in this config?  Using a UUID algorithm to generate a unique ID makes sense, but once it's generated I think we can treat it as an opaque ID and won't need access to any `UUID`-specific methods.